cli/src/opts/cast.rs

```diff
@@ -139,6 +139,8 @@ pub enum Subcommands {
         cast_async: bool,
         #[structopt(flatten)]
         eth: EthereumOpts,
+        #[structopt(long, help="use the flashbots RPC URL")]
+        flashbots: bool,
```

Comment:
Let's maybe pull `flashbots` inside `EthereumOpts` and make `eth.rpc_url` an `Option<String>`, then add a helper function `eth.rpc_url() -> Result<&str>` which does:

```
if flashbots {
      Ok(FLASHBOTS_URL)
} else {
     self.rpc_url.ok_or(|| eyre::Error::msg("no Ethereum RPC provided, maybe you forgot to set the --rpc-url or the ETH_RPC_URL parameter? Alternatively, consider using the --flashbots flag to get frontrunning protection")
}